2480 - Fix public.migrations column timestamp and delete old migration entries from table

sorja opened the pull request:
Add by default timestamp so we don't have to add it from outside
It is now consistent with migration_steps

sorja committed:
2480 - fix public.migrations run_on timestamp, set default now

sorja committed:
fix migration step typo

minotogna commented on src/test/migrations/steps/20240912102424-step-fix-migrations-table-ddl.ts:
```diff
@@ -0,0 +1,10 @@
+  await client.query(`
+    alter table migrations
+    alter column run_on set default now(),
+    alter column run_on type timestamp without time zone
```
Since we haven't merged this yet, I wouldn't add a step , but update only initial step . What do u think ?

sorja replied:
Unfortunately we have this table already in public schema
![image](https://github.com/user-attachments/assets/29d0f656-880d-4f22-8860-a524d619883c)


We reuse the same table that we had for the old migrations system, but we will empty it

What do you suggest?

minotogna replied:
Update the initial step ? Didn't you do this already ?

minotogna replied:
oh I get it now. 
this step should run before inserting the migrations in this table for current applications? or, can we do it in the same migration step (event better)?

sorja replied:
Current status:
We have `public.migrations` table, that has no default value when inserting new rows
The idea of this step is to update the existing `migrations` table to be up to date with our init public schema, which has updated version of the `migrations` table.

So we are only updating what we already have in DB, hope this makes sense!

Let me know

minotogna replied:
yes, we need one migration step that operates on current applications. ok ?

sorja replied:
It is this one introduced in this PR?
`src/test/migrations/steps/20240912102424-step-fix-migrations-table-ddl.ts`

Sorry, maybe I am not following!

mergify[bot] committed:
Merge branch '2480-setup-public-database-migrations' into 2480-setup-public-database-migrations_fix-migrations-table-ddl-default

mergify[bot] committed:
Merge branch '2480-setup-public-database-migrations' into 2480-setup-public-database-migrations_fix-migrations-table-ddl-default

mergify[bot] committed:
Merge branch '2480-setup-public-database-migrations' into 2480-setup-public-database-migrations_fix-migrations-table-ddl-default

sorja commented:
Note:
Single migration step to:
- update `public.migrations`
- insert lines in `public.migrations` for migrations we dont wanna run in our existing instances

sorja committed:
test.js.yml: fix command order

first run build
then run migrations

sorja committed:
2480 - Remove unnecessary migration steps

Remove unnecessary migration steps
Merge the content to step-exclude-public-migrations

sorja commented on .github/workflows/test.js.yml:
```diff
@@ -55,14 +55,14 @@ jobs:
             NODE_AUTH_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```
Restore the original order we had.
First run migrations and then build the application

I don't think it matters in practice, but it makes more sense to be in this original order.
